linear_operator/operators/interpolated_linear_operator.py

```diff
@@ -454,3 +455,28 @@ def zero_mean_mvn_samples(
         res = left_interp(self.left_interp_indices, self.left_interp_values, base_samples).contiguous()
         batch_iter = tuple(range(res.dim() - 1))
         return res.permute(-1, *batch_iter).contiguous()
+
+    def to(self: Float[LinearOperator, "*batch M N"], *args, **kwargs) -> Float[LinearOperator, "*batch M N"]:
+
+        # Overwrite the to() method in _linear_operator to avoid converting index matrices to float.
+        # Will only convert both dtype and device when arg and dtype are both int/float.
+        # Otherwise, will only convert device.
+
+        device, dtype = _to_helper(*args, **kwargs)
+
+        new_args = []
+        new_kwargs = {}
+        for arg in self._args:
+            if hasattr(arg, "to"):
+                if arg.dtype.is_floating_point == dtype.is_floating_point:
```

Comment:
It could be that `arg` has a `to` attribute but doesn't have a `dtype` attribute (e.g. if it is some other non-tensor or non-module object that just happens to have a `to` method). Not sure exactly how we'd handle this but we probably should? Maybe we want to catch and pass on exceptions here?

Reply:
That's a good point. Maybe I can add another condition to first check if the `arg` has a `dtype` attribute, before the `is_floating_point` check? If the `arg` does not have a `dtype` attribute, we directly jump to the `device` conversion.